### PR TITLE
[docs] Add docs for mongodb plugin

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 1]
```diff
@@ -1 +1,121 @@
...
+    sort: { age: 1 },
+    projection: { age: 1, name: 1, _id: 0 },
+    batchSize: 500
+}
+` against `users` database on `mock` MongoDB instance and save result to SCENARIO variable `find`
+----
+
+=== Execute chained commands
+
+Executes assembled sequece of predefined commands and performs them against mongo database and saves the result to a variable
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:61]: `sequece` -> sequence

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 2]
```diff
@@ -1 +1,121 @@
...
+----
+
+* `$commands` - Sequence of commands to execute
+* `$collectionName` - Collection name to retrieve documents from
+* `$dbName` - Database name
+* `$instanceKey` - Key of particular connection under `mongodb.connection.` prefix
+* `$scopes` - xref:parameters:variable-scope.adoc[The comma-separated set of the variables scopes].
+* `$variableName` - The variable name to store results in JSON format.
+
+.Find employees who is older than 20 years
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:112]: are

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 3]
```diff
@@ -1 +1,121 @@
...
+When I execute command `$command` against `$dbName` database on `$instanceKey` MongoDB instance and save result to $scopes variable `$variableName`
+----
+
+* `$command` - Native command to perform
+* `$dbName` - Database name
+* `$instanceKey` - Key of particular connection under `mongodb.connection.` prefix
+* `$scopes` - xref:parameters:variable-scope.adoc[The comma-separated set of the variables scopes].
+* `$variableName` - The variable name to store results in JSON format.
+
+.Find employees who is older than 20 years
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:45]: https://github.com/vividus-framework/vividus/pull/1346/files#r558216940

**Review by ikalinin1 — APPROVED** [review]

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 4]
```diff
@@ -1 +1,121 @@
...
+implementation(group: 'org.vividus', name: 'vividus-plugin-mongodb', version: '{current-version}')
+----
+
+== Properties
+
+It's allowed to configure unlimited number of MongoDB connections via mechanism of the dynamic properties. The properties prefix example is:
+```properties
+mongodb.connection.mock=mongodb+srv://username:password@mock-zzw4a.mongodb.net/test?retryWrites=true&w=majority
+```
+where `mock` is a key used to refer MongoDB connection in the steps. The key is defined by users, must be unique and can't contain dots.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:19]: `mock` -> `my-mongodb-instance` ?

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 5]
```diff
@@ -1 +1,121 @@
...
+
+=== Execute native command
+
+Executes a mongo native command against a mongo database and saves command's result to a variable
+
+[NOTE]
+====
+* https://docs.mongodb.com/manual/reference/command/[Native commands]
+* database in query is not required to be the same as the authentication database
+* does not support the database cursors logic, so if you use find command to query more than 101 document then adjust `batchSize` property according to the number of documents being queried
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:31]: Use either Capital letter in upper or lower case in all list items

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 6]
```diff
@@ -1 +1,121 @@
...
+
+== Steps
+
+=== Execute native command
+
+Executes a mongo native command against a mongo database and saves command's result to a variable
+
+[NOTE]
+====
+* https://docs.mongodb.com/manual/reference/command/[Native commands]
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:29]: give short description what native commands are

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 7]
```diff
@@ -1 +1,121 @@
...
+* database in query is not required to be the same as the authentication database
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:30]: please elaborate

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 8]
```diff
@@ -1 +1,121 @@
...
+* does not support the database cursors logic, so if you use find command to query more than 101 document then adjust `batchSize` property according to the number of documents being queried
+====
+
+[source,gherkin]
+----
+When I execute command `$command` against `$dbName` database on `$instanceKey` MongoDB instance and save result to $scopes variable `$variableName`
+----
+
+* `$command` - Native command to perform
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:39]: `to perform` -> `to execute`

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 9]
```diff
@@ -1 +1,121 @@
...
+* `$dbName` - Database name
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:40]: The database name

File: docs/modules/plugins/pages/plugin-mongodb.adoc (hunk 8 above)
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:39]: The native command

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 10]
```diff
@@ -1 +1,121 @@
...
+* `$instanceKey` - Key of particular connection under `mongodb.connection.` prefix
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:41]: The key of the ...

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 11]
```diff
@@ -1 +1,121 @@
...
+|terminal
+|collects previously found documents into JSON format, takes no arguments
+|
+
+|===
+
+[NOTE]
+====
+* command sequence must start with `source` command followed by any number of `intermediate` commands and end with `terminal` command
+* supports database cursors logic
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:97]: please elaborate

**Review by valfirst — CHANGES_REQUESTED** [review]
Please review all the docs and put missing articles

File: docs/modules/plugins/pages/plugin-mongodb.adoc (hunk 6 above)
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-mongodb.adoc:29]: renamed to database command as per docs

File: docs/modules/plugins/pages/plugin-mongodb.adoc (hunk 7 above)
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-mongodb.adoc]: forgot what I meant by this :D removed

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 12]
```diff
@@ -1 +1,119 @@
-include::ROOT:page$under-construction.adoc[]
+= MongoDB Plugin
+
+The plugin provides a functionality to interact with the https://www.mongodb.com/[MongoDB].
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:3]: ~the~

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 13]
```diff
@@ -1 +1,119 @@
...
+```properties
+mongodb.connection.my-mongodb-instance=mongodb+srv://username:password@mock-zzw4a.mongodb.net/test?retryWrites=true&w=majority
+```
+where `my-mongodb-instance` is a key used to refer a MongoDB connection in the steps. The key is defined by users, must be unique and can't contain dots.
+
+== Steps
+
+=== Execute database command
+
+Executes a mongo database command against a mongo database and saves the command's result to a variable
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:25]: `the command result`

File: docs/modules/plugins/pages/plugin-mongodb.adoc [hunk 14]
```diff
@@ -1 +1,119 @@
...
+
+[NOTE]
+====
+* https://docs.mongodb.com/manual/reference/command/[Database commands]
+* For the `find` command the number of documents returned from the database is limited by the `batchSize` property that defaults to 101, so if you expect a dataset larger that the default value then adjust the `batchSize` property according to the number of expected documents
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-mongodb.adoc:30]: `... larger than the default ...`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**